waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -1,32 +1,45 @@
 module Wasp.Project.Analyze
   ( analyzeWaspProject,
-    readPackageJsonFile,
     analyzeWaspFileContent,
     findWaspFile,
     findPackageJsonFile,
     analyzePrismaSchema,
+    WaspFile (..),
   )
 where
 
+import Control.Applicative ((<|>))
 import Control.Arrow (ArrowChoice (left))
+import Control.Concurrent (newChan)
+import Control.Concurrent.Async (concurrently)
+import Control.Monad.Except (ExceptT (..), runExceptT)
 import qualified Data.Aeson as Aeson
+import Data.Conduit.Process.Typed (ExitCode (..))
 import Data.List (find, isSuffixOf)
-import StrongPath (Abs, Dir, File', Path', toFilePath, (</>))
+import StrongPath (Abs, Dir, File', Path', Rel, basename, toFilePath, (</>))
+import qualified StrongPath as SP
+import StrongPath.TH (relfile)
+import StrongPath.Types (File)
 import qualified Wasp.Analyzer as Analyzer
 import Wasp.Analyzer.AnalyzeError (getErrorMessageAndCtx)
 import Wasp.Analyzer.Parser.Ctx (Ctx)
 import qualified Wasp.AppSpec as AS
+import Wasp.AppSpec.Core.Decl.JSON ()
 import Wasp.AppSpec.PackageJson (PackageJson)
 import qualified Wasp.AppSpec.Valid as ASV
 import Wasp.CompileOptions (CompileOptions)
 import qualified Wasp.CompileOptions as CompileOptions
 import qualified Wasp.ConfigFile as CF
 import Wasp.Error (showCompilerErrorForTerminal)
 import qualified Wasp.Generator.ConfigFile as G.CF
+import qualified Wasp.Generator.Job as J
+import Wasp.Generator.Job.IO (readJobMessagesAndPrintThemPrefixed)
+import Wasp.Generator.Job.Process (runNodeCommandAsJob)
```

Comment:
You're right, nice catch. I didn't notice it was there.

And there's nothing generator-specific about it. I think it's just there because the generator used to be the only one running jobs. I'll move it somewhere more appropriate.

Reply:
Leaving this one for later.

Reply:
Done in #2407